packages/arb-node-core/staker/staker.go

```diff
@@ -210,10 +223,27 @@ func (s *Staker) Act(ctx context.Context) (*arbtransaction.ArbTransaction, error
 		return nil, err
 	}
 	if !nodesLinear {
-		logger.Warn().Msg("Fork detected")
+		logger.Warn().Msg("fork detected")
 		if effectiveStrategy == DefensiveStrategy {
 			effectiveStrategy = StakeLatestStrategy
 		}
+		s.inactiveLastCheckedNode = nil
+	}
+	if s.bringActiveUntilNode != nil {
+		if info.LatestStakedNode.Cmp(s.bringActiveUntilNode) < 0 {
+			if effectiveStrategy == DefensiveStrategy {
+				effectiveStrategy = StakeLatestStrategy
+			}
+			s.inactiveLastCheckedNode = nil
+		} else {
+			logger.Info().Msg("defensive validator staked past incorrect node; waiting here")
+			s.bringActiveUntilNode = nil
+			s.inactiveLastCheckedNode = nil
```

Comment:
Probably clearer to lift this out of the if statement since it's the same in both clauses